### ISSUE TITLE
Preventing file upload for files over 20mb still renders an empty thumbnail preview

### ISSUE THREAD

**Issue by alexlopezit** [GetStream/stream-chat-react#595]
We are currently trying to prevent the user from selecting images bigger than 20mb as it looks like is the limit on the API.

I tried to use the `doImageUploadRequest` where I can check the size of the image, the idea is to notify the user when the file is bigger than 20mb and then not upload that image, here is the code I am using.

```
  const doImageUploadRequest = (file: File) => {
    const { size } = file
    const sizeInMB = size / (1024 * 1024)

    if (sizeInMB > 20) {
      // We can show a toast notification here letting the user know
      return channel.sendImage(null)
    }

    return channel.sendImage(file)
  }

...

<MessageInput doImageUploadRequest={doImageUploadRequest} />
```

The problem is that even when I try to not upload that large image, a thumbnail preview is generated which contains a retry svg icon, is there a way to prevent the rendering of this "retry thumbnail"?

cc @tbarbugli 

gz#6954

**Comment by jaapbakker88** [comment]
@alexlopezit we've just released version `3.1.7` which will remove the preview if the `doImageUploadRequest` returns a falsy value.

Let me know if that works for you

**Comment by alexlopezit** [comment]
Thanks a lot @jaapbakker88 , that fixed this issue
